src/wp-includes/class-wp-theme-json.php

```diff
@@ -855,7 +855,7 @@ protected static function get_blocks_metadata() {
 				static::$blocks_metadata[ $block_name ]['elements'][ $el_name ] = implode( ',', $element_selector );
 			}
 			// If the block has style variations, append their selectors to the block metadata.
-			if ( ! empty( $block_type->styles ) ) {
+			if ( ! empty( $block_type->styles ) && is_array( $block_type->styles ) ) {
```

Comment:
This can be simplified as follows:

```suggestion
			if ( is_array( $block_type->styles ) ) {
```

Then if the styles `array` is empty, the `foreach()` does not iterate.

This check though should be unnecessary and could be removed once there is validation added to `WP_Block_Type` for setting properties.